profiler/profile.go

```diff
@@ -216,6 +229,108 @@ func goroutineProfile(cfg *config) (*profile, error) {
 	}, nil
 }
 
+func goroutineWaitProfile(cfg *config) (*profile, error) {
+	var (
+		text  = &bytes.Buffer{}
+		pprof = &bytes.Buffer{}
+		start = now()
+	)
+	if err := lookupProfile(GoroutineProfile.String(), text, 2); err != nil {
+		return nil, err
+	} else if err := goroutineDebug2ToPprof(text, pprof); err != nil {
+		return nil, err
+	}
+	end := now()
+	tags := append(cfg.tags, GoroutineWaitProfile.Tag())
+	cfg.statsd.Timing("datadog.profiler.go.collect_time", end.Sub(start), tags, 1)
+
+	return &profile{
+		name: GoroutineWaitProfile.Filename(),
+		data: pprof.Bytes(),
+	}, nil
+}
+
+func goroutineDebug2ToPprof(r io.Reader, w io.Writer) error {
+	goroutines, errs := stackparse.Parse(r)
+
+	functionID := uint64(1)
+	locationID := uint64(1)
+
+	p := &pprofile.Profile{}
+	m := &pprofile.Mapping{ID: 1, HasFunctions: true}
+	p.Mapping = []*pprofile.Mapping{m}
+	p.SampleType = []*pprofile.ValueType{
+		{
+			Type: "waitduration",
+			Unit: "nanoseconds",
+		},
+	}
+
+	for _, g := range goroutines {
+		// TODO(fg) exclude goroutines w/ g.Wait == 0?
```

Comment:
@pmbauer what do you think? Our backend/ui doesn't show this data right now anyway, so it would reduce the profile size.

Reply:
since these are surfaced in the regular goroutine profile I think we can exclude?

Reply:
They are in the regular profile, but without goroutine ids and state.

With the current visualization I can't think of a good reason to keep them here. But here is a future use case I could imagine: Let's say you got 4 workers, and 2 of them are "stuck" (in waiting state). If we have the non-stock goroutines in the profile, we could display this info to the user somehow.

So maybe let's keep this for now?

Reply:
Excellent point. Let's keep it.